livy_submit/cli.py

```diff
@@ -403,7 +403,7 @@ def _livy_submit_parser(subparsers):
     ap.add_argument(
         "--name",
         action="store",
-        required=True,
+        required=False,
         help="The name that your Spark job should have on the Yarn RM",
```

Comment:
What does the Yarn RM say is the "Name" if we don't provide a name? I'd like to document that so we know for future us. Whatever we discover should also be documented in the LivyAPI.submit() function.

Reply:
on the resource manager the Name is given as the name of the file and can appear multiple times. From the LivyAPI name will return as JSON `null`, which is parsed to the singleton `None`.

Reply:
added to docstring